### PR TITLE
Check that property is set in the conversion methods

### PR DESCRIPTION
**Pull request by clpetix**

### CODE REVIEW

**Comment by clpetix** [comment]
@mphoward this should address the issues that I had when using the GSD frame directly. The `to_hoomd_gsd` already checked all of the properties with `has_XX()`methods. 

Once this is good to go, then PR #54 should also be ready for review!

File: src/lammpsio/snapshot.py [hunk 1]
```diff
@@ -84,14 +84,20 @@ def from_hoomd_gsd(cls, frame):
...
+        if frame.particles.typeid is not None:
+            snap.typeid = frame.particles.typeid + 1
+        if frame.particles.charge is not None:
+            snap.charge = frame.particles.charge
+        if frame.particles.mass is not None:
+            snap.mass = frame.particles.mass
+        if frame.particles.body is not None:
+            snap.molecule = frame.particles.body + 1
         if numpy.any(snap.molecule < 0):
             warnings.warn("Some molecule IDs are negative, remapping needed.")
```
Comment (mphoward) [inline src/lammpsio/snapshot.py:102]: Move this inside the `if body is not None` check, it only needs to run if this field gets set

File: src/lammpsio/snapshot.py [hunk 2]
```diff
@@ -84,14 +84,20 @@ def from_hoomd_gsd(cls, frame):
...
-        snap.position = frame.particles.position
-        snap.velocity = frame.particles.velocity
-        snap.image = frame.particles.image
-        snap.typeid = frame.particles.typeid + 1
-        snap.charge = frame.particles.charge
-        snap.mass = frame.particles.mass
-
-        snap.molecule = frame.particles.body + 1
+        if frame.particles.position is not None:
+            snap.position = frame.particles.position
```
Comment (mphoward) [inline src/lammpsio/snapshot.py:88]: Please add whitespace between these to improve readability
```suggestion
            snap.position = frame.particles.position

```

**Review by mphoward — CHANGES_REQUESTED** [review]
Overall this looks straightforward and makes sense! Please add a unit test that would fail before the change was applied and passes now. It could be as straightforward as passing an almost empty `Frame`

**Comment by clpetix** [comment]
@mphoward I added a unit test that makes a snapshot from a minimal GSD and then checks that snap.molecule initializes to the default zeros when called. I also confirmed that this test previously failed before these changes. 

This should be good for you to look at again!

**Review by mphoward — CHANGES_REQUESTED** [review]
Thanks! I have one request to tweak the test and then this can be merged.

**Comment by clpetix** [comment]
@mphoward , that makes sense. This should be good now!

**Review by mphoward — APPROVED** [review]
Thank you!

**Pull request merged**